src/ctelnet.cpp

```diff
@@ -407,9 +407,11 @@ void cTelnet::handle_socket_signal_connected()
     QString nothing = "";
     mpHost->mLuaInterpreter.call(func, nothing);
     mConnectionTime.start();
-    if ((mpHost->getLogin().size() > 0) && (mpHost->getPass().size() > 0)) {
```

Comment:
A niggle about the execution: You are lumping two changes together again :frowning_face:

Reply:
I saw it as an error in the same functional part of the code - it wouldn't allow the user to set just a character name and for that to be sent without a password being also provided. This change makes things adopt a position with a lower amount of surprise. E.g. you enter just a character name - just that gets sent; you enter both a character name and a password ***then*** both get sent; entering *just* a password is plainly weird and there is no change in how that works out... :stuck_out_tongue_winking_eye:

With the recent addition of the "don't paint my password up on the main console" bugfix (#2236) it could be an option to provide just the character name and to type in the password in the command line (or send it via a normal `send(...)` in an alias or something...)

If you feel ***that*** strongly I will put the changes in this file in a separate PR which will probably go first and then leave the other lua stuff to be discussed/rationalised/aborted in this PR.

Reply:
I don't feel that strongly about including it here I just wanted to give a reminder :wink:.
But at the same time, I am not completely sure it's an error either and in a separate PR we could have this discussion a bit more streamlined than having it intertwined with another non-trivial discussion.